### PR TITLE
DSPDC-1149 Add release_date column to every table

### PR DESCRIPTION
**Pull request by danxmoran**
And thread it through the pipeline.

I added an unrelated unit-test to make myself feel better about the coverage.

### CODE REVIEW

File: schema/src/main/jade-tables/clinical_assertion.table.json [hunk 1]
```diff
@@ -6,6 +6,11 @@
       "datatype": "string",
       "type": "primary_key"
     },
+    {
+      "name": "release_date",
```
Comment (danxmoran) [inline schema/src/main/jade-tables/clinical_assertion.table.json:10]: Note to self: Need to sweep through these tables and change the partition strategy to use this column

Comment (quazi-broad) [reply schema/src/main/jade-tables/clinical_assertion.table.json:10]: Where would that change need to be made?

Comment (danxmoran) [reply schema/src/main/jade-tables/clinical_assertion.table.json:10]: It's in the same file, will comment after I push the changes

File: schema/src/main/jade-tables/clinical_assertion.table.json [hunk 2]
```diff
@@ -157,5 +162,9 @@
       "struct_name": "interpretation_comment",
       "type": "repeated"
     }
-  ]
+  ],
+  "partitioning": {
+    "mode": "date_from_column",
+    "column": "release_date"
+  }
```
Comment (danxmoran) [inline schema/src/main/jade-tables/clinical_assertion.table.json:169]: @quazi-broad like this

Comment (snwessel) [reply schema/src/main/jade-tables/clinical_assertion.table.json:169]: Thanks for highlighting that! It's more straightforward than I was expecting.

**Pull request merged**